pkg/local_workflows/code_workflow/native_workflow.go

```diff
@@ -0,0 +1,172 @@
+	codeclienthttp "github.com/snyk/code-client-go/http"
+	"github.com/snyk/code-client-go/sarif"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/content_type"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/json_schemas"
+	"github.com/snyk/go-application-framework/pkg/utils"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+)
+
+const (
+	summaryType        = "sast"
+	SARIF_CONTENT_TYPE = "application/sarif+json"
+)
+
+type OptionalAnalysisFunctions func(string, func() *http.Client, *zerolog.Logger, configuration.Configuration) (*sarif.SarifResponse, error)
+
+func EntryPointNative(invocationCtx workflow.InvocationContext, opts ...OptionalAnalysisFunctions) ([]workflow.Data, error) {
+	// get necessary objects from invocation context
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	id := invocationCtx.GetWorkflowIdentifier()
+
+	path := config.GetString(configuration.INPUT_DIRECTORY)
+
+	logger.Debug().Msgf("Path: %s", path)
+
+	analyzeFnc := defaultAnalyzeFunction
+	if len(opts) == 1 {
+		analyzeFnc = opts[0]
+	}
+
+	result, err := analyzeFnc(path, invocationCtx.GetNetworkAccess().GetHttpClient, logger, config)
+	if err != nil || result == nil {
+		return nil, err
+	}
+
+	sarifData, err := createCodeWorkflowData(workflow.NewTypeIdentifier(id, "sarif"), &result.Sarif, SARIF_CONTENT_TYPE, path)
+	if err != nil {
+		return nil, err
+	}
+
+	summary := createCodeSummary(&result.Sarif)
+	summaryData, err := createCodeWorkflowData(workflow.NewTypeIdentifier(id, "summary"), summary, content_type.TEST_SUMMARY, path)
+	if err != nil {
+		return nil, err
+	}
+
+	return []workflow.Data{sarifData, summaryData}, nil
+}
+
+func defaultAnalyzeFunction(path string, httpClientFunc func() *http.Client, logger *zerolog.Logger, config configuration.Configuration) (*sarif.SarifResponse, error) {
+	var result *sarif.SarifResponse
+
+	interactionId, err := uuid.GenerateUUID()
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Debug().Msgf("Interaction ID: %s", interactionId)
+
+	files, err := getFilesForPath(path)
+	if err != nil {
+		return nil, err
+	}
+
+	changedFiles := make(map[string]bool)
+	ctx := context.Background()
+	codeInstrumentor := &codeClientInstrumentor{}
+	codeErrorReporter := &codeClientErrorReporter{}
+	httpClient := codeclienthttp.NewHTTPClient(httpClientFunc, codeclienthttp.WithLogger(logger))
+	codeScannerConfig := &codeClientConfig{
+		localConfiguration: config,
+	}
+	codeScanner := codeclient.NewCodeScanner(httpClient, codeScannerConfig, codeInstrumentor, codeErrorReporter, logger)
+	result, _, err = codeScanner.UploadAndAnalyze(ctx, interactionId, path, files, changedFiles)
+	return result, err
+}
+
+// todo: recursively iterate and filter
+func getFilesForPath(path string) (<-chan string, error) {
+	filter := utils.NewFileFilter(path)
+	rules, err := filter.GetRules([]string{})
```

Comment:
question: we're not supplying any rule files here so it will only ignore `.git` files by default.